core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -267,17 +269,19 @@ class ReplicaFetcherThread(name: String,
     }
   }
 
+  def withinQuota(partition: TopicAndPartition): Boolean = {
+    !(quota.isThrottled(partition) && quota.isQuotaExceeded)
+  }
+
   protected def buildFetchRequest(partitionMap: Map[TopicAndPartition, PartitionFetchState]): FetchRequest = {
     val requestMap = mutable.Map.empty[TopicPartition, JFetchRequest.PartitionData]
-
-    partitionMap.foreach { case ((TopicAndPartition(topic, partition), partitionFetchState)) =>
-      if (partitionFetchState.isActive)
-        requestMap(new TopicPartition(topic, partition)) = new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize)
+    partitionMap.foreach { case ((partition, partitionFetchState)) =>
+      if (partitionFetchState.isActive && withinQuota(partition))
```

Comment:
Would it be better to just check quota.isQuotaExceeded once and make the same decision for each throttled partition on whether it should be included or not?